src/bio/bio_xstream.c

```diff
@@ -400,7 +400,10 @@ bio_nvme_init(const char *nvme_conf, int shm_id, int mem_size,
 	nvme_glb.bd_bs_opts.max_channel_ops = BIO_BS_MAX_CHANNEL_OPS;
 
 	bio_chk_cnt_init = DAOS_DMA_CHUNK_CNT_INIT;
-	bio_chk_cnt_max = DAOS_DMA_CHUNK_CNT_MAX;
+	if (mem_size != DAOS_NVME_MEM_PRIMARY)
```

Comment:
Shouldn't caller always provide an valid "mem_size"? And we do need to check if the provided "mem_size", if it's too small (for example, if it's not enough for 1GB per target), we'd fail the bio_nvme_init().

Reply:
do we want to fail bio_nvme_init() if the mem_size is too small? right now in the code if the mem_size provided by the user is too small, than the current default 1GB per target is used.